client/lib/sites-list/list.js

```diff
@@ -394,7 +394,7 @@ SitesList.prototype.setRecentlySelectedSite = function( siteID ) {
 };
 
 SitesList.prototype.getRecentlySelected = function() {
-	if ( ! this.recentlySelected.length ) {
+	if ( ! this.recentlySelected.length || ! this.initialized ) {
```

Comment:
I see, so the issue is that we may have values in `recentlySelected` prior to having values in `this.get()`, which we filter upon later in this function. This makes sense to me.